pkg/controller/extendeddaemonsetreplicaset/strategy/utils.go

```diff
@@ -23,9 +23,29 @@ import (
 	datadoghqv1alpha1 "github.com/datadog/extendeddaemonset/pkg/apis/datadoghq/v1alpha1"
 	"github.com/datadog/extendeddaemonset/pkg/controller/extendeddaemonsetreplicaset/conditions"
 	podaffinity "github.com/datadog/extendeddaemonset/pkg/controller/utils/affinity"
+	"github.com/datadog/extendeddaemonset/pkg/controller/utils/comparison"
 	podutils "github.com/datadog/extendeddaemonset/pkg/controller/utils/pod"
 )
 
+func compareCurrentPodWithNewPod(edsName string, replicaset *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet, pod *corev1.Pod, node *corev1.Node) bool {
+	// check that the pod correspond to the replicaset. if not return false
+	if !compareSpecTemplateMD5Hash(replicaset.Spec.TemplateGeneration, pod) {
```

Comment:
should we check for nil pointers here ?

Reply:
on the `replicaset`? It is a mandatory args. and also the function is private. So I think it is fine to not check if it nil. wdyt?

Reply:
okay